### DISCUSSION TITLE
Exception Clarity Improvement

### DISCUSSION THREAD

**Discussion by tofanpetru** [MapsterMapper/Mapster#648] [Q&A]
Hello everyone! 

I would like to receive some advice. How can I make the result of an exception much clearer? For example, what values and properties caused the exception? 

The ultimate goal is to log this message to be much more user-friendly and clear. Because the application is running somewhere and it's very difficult to investigate

Let's assume that Mapster will throw an exception if it performs an incorrect conversion.
![img1](https://github.com/MapsterMapper/Mapster/assets/13838191/cd3ec2ef-88a5-4caa-a4d6-968c3f61999f)

**Comment by DocSvartz** [comment]
Hello @tofanpetru. 
Based on the code I saw while working on current Issue. The Mapster can throw only 2 types of exceptions:
1) If it is not possible to create a Conversion function based on the passed Types.
2) If it is also impossible to create such a Conversion function for a surrogate Type (Mapping to Inteface).

The error you showed occurs after the main work, outside of the Mapster Code itself.
This is the result of calling the conversion function.

Сan wrap in try-catch blocks. The place where the function itself is called is in the adapter. But then you will only receive data about the top-level types transferred for processing.

```
var result = MyDto.Adapt<MyClass>();

MyClass // You will receive information that everything broke when converting  MyDto map to MyClass
{
  MyValueType<decmal> // fail parse
}
```


Such  try-catch blocks are not created in the Conversion function itself.

**Reply by DocSvartz** [reply]
@tofanpetru  You use dependency injection ?

**Reply by tofanpetru** [reply]
@DocSvartz I use Dependency Injection. Thank you very much for the advice. I will try to create an elegant method to extract error messages

**Reply by DocSvartz** [reply]
Does the current depth of the stack trace go to the actual place where the converter is called in your code?
Or only to the "factory" where these injections are created?
I was just testing the manual call of the maper)

**Reply by tofanpetru** [reply]
@DocSvartz This is up to the point where it was called

**Reply by DocSvartz** [reply]
Ок. I understand, I just thought that this was also the problem.
